### PR TITLE
Different decimal place for crypto and fiat currency

### LINKED ISSUE #3287: [BUG] 4 decimal is not enough for crypto
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

image

The exchange rate cannot have more than 4 decimals, so when exchanging JPY for BTC (1 JPY = 0.00000011 BTC), it cannot work correctly.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Screenshots

_No response_

### App version

2024.07.06 (181)

### Smartphone

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by suyash01**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [ ] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?
Describe with a few bullets **what's new:**

- Updated to use different decimal place for crypto and fiat currency

|Before|After|
:------:|:-----:
  |  
  |  
  |  

## Risk factors

**What may go wrong if we merge your PR?**

- Exchange rate wrapping may happen for small screen

**In what cases won't your code work?**

- ...
- ...

## Does this PR close any GitHub issues?

- Closes #3287 

## Troubleshooting CI failures ❌

Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: temp/legacy-code/src/main/java/com/ivy/legacy/domain/data/IvyCurrency.kt [hunk 1]
```diff
@@ -208,6 +211,13 @@ data class IvyCurrency(
         fun getDefault(): IvyCurrency = IvyCurrency(
             fiatCurrency = getDefaultFIATCurrency()
         )
+
+        fun getDecimalPlaces(code: String): Int {
+            if (code.isBlank()) return 0
+
+            val crypto = CRYPTO.find { it.code == code }
```
[the remark this answers is not in the document]
Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/domain/data/IvyCurrency.kt:218]: Even simpler:
```kotlin
fun getDecimalPlaces(assetCode: String): Int = if (assetCode in CRYPTO) CRYPTO_... else FIAT...
```

**Review by ILIYANGERMANOV — COMMENTED** [review]
LGTM, left a few comments

**Comment by ILIYANGERMANOV** [comment]
@suyash01 can you test if the Amount input keyboard works with crypto? I'd appreciate a screen recording where you make a crypto transaction

File: temp/legacy-code/src/main/java/com/ivy/legacy/domain/data/IvyCurrency.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (suyash01) [reply temp/legacy-code/src/main/java/com/ivy/legacy/domain/data/IvyCurrency.kt:218]: Did you mean like this:
```kotlin
fun getDecimalPlaces(assetCode: String): Int = if(fromCode(assetCode) in CRYPTO) CRYPTO_DECIMAL else FIAT_DECIMAL
```

[the remark this answers is not in the document]
Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/domain/data/IvyCurrency.kt:218]: yup

**Comment by suyash01** [comment]
https://github.com/Ivy-Apps/ivy-wallet/assets/18034836/1611b613-8ab6-469f-8cf9-260f7a930338

@ILIYANGERMANOV seems to be working fine with crypto as both to and from

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**